docs/tutorial.rst

```diff
@@ -306,17 +318,281 @@ trees. In this example we incorrectly inferred that 0 coalesces with the
 ancestor of nodes 2, 3, 4 and 5 before 1.
 
 
-.. todo::
+************
+Data example
+************
+
+Inputting real data for inference is similar in principle to the examples above:
+you simply iterate over the sites in your data file, calling :meth:`.SampleData.add_site`
+for each variable site. We do not provide methods to read directly from other formats, as
+these formats are often too complex for automatic conversion. However, many Python
+libraries are available for reading different genomic data formats, and writing a script
+to iterate over the sites in such files is usually a fairly simple task. The example
+below shows how this can be done for a VCF file, using a freely available VCF reading
+library.
+
++++++++++++++
+Reading a VCF
++++++++++++++
+
+A common way to store genetic variation data is as a VCF (Variant Call Format) file.
+The `CYVCF2 library <https://github.com/brentp/cyvcf2>`_ provides a relatively fast
+Python interface for reading VCF files. The following script demonstrates how to infer a
+tree sequence from a simple VCF file: in particular the function below named
+``add_diploid_sites`` illustrates how to iterate over the variants in a ``CYVCF2.VCF``
+object and add them to a ``tsinfer`` sample data file. This particular script assumes
+that the VCF describes a set of phased, diploid individuals, and has the ancestral state
+present in the INFO field of the VCF (otherwise it takes the reference allele as the
+ancestral state). As example data, we use a publicly available VCF file of the genetic
+variants from chromosome 24 of ten Norwegian and French house sparrows,
+*Passer domesticus* (thanks to Mark Ravinet for the data file):
+
+.. code-block:: python
+
+    import cyvcf2
+    import tsinfer
+    
+    def add_diploid_sites(vcf, samples):
+        pos = None
+        for variant in vcf:
+            if pos == variant.POS:
+                raise ValueError("Duplicate positions for variant at position", pos)
+            else:
+                pos = variant.POS
+            if any([row[2]==False for row in variant.genotypes]):
```

Comment:
Comparing things to False is a bad idea. A more idiomatic thing to do is ``any(not row[2] for row in ...]). It's not obvious why this would mean it's unphased though. An explanation or a link to the docs would be good (people will ask us questions otherwise...)

Reply:
If cyvcf2 just returns a tuple, then something like
``any(not phased for _, _, phased in variant.genotypes)`` would be better. Hopefully it returns a namedtuple or something though?

Reply:
Sadly, in the version of cyvcf2 that I have, the `.genotypes` attribute for a set of N diploid samples simply contains an N length list of size 3 lists. I'll use your `_,_,phased` notation, thanks.